### PR TITLE
Games Master Cartridge slot device for the Color Computer

### PR DESCRIPTION
**Pull request by tlindner**
Adds a new slot device to emulate the Games Master Cartridge developed by John Linville.

### CODE REVIEW

File: src/devices/bus/coco/coco_gmc.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// license:BSD-3-Clause
+// copyright-holders:tim lindner
+/***************************************************************************
+
+    coco_gmc.c
```
Comment (npwoods) [inline src/devices/bus/coco/coco_gmc.cpp]: coco_gmc.cpp :-)

File: src/devices/bus/coco/coco_gmc.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//-------------------------------------------------
+
+machine_config_constructor coco_pak_gmc_device::device_mconfig_additions() const
+{
+	return MACHINE_CONFIG_NAME( cocopak_gmc );
+}
+
+/*-------------------------------------------------
+    write
+-------------------------------------------------*/
```
Comment (npwoods) [inline src/devices/bus/coco/coco_gmc.cpp]: Use C++ style comments

File: src/devices/bus/coco/coco_gmc.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// license:BSD-3-Clause
+// copyright-holders:tim lindner
+#pragma once
+
+#ifndef __COCO_GMC_H__
+#define __COCO_GMC_H__
```
Comment (npwoods) [inline src/devices/bus/coco/coco_gmc.h]: Symbols starting with underscores are bad; make this MAME_DEVICES_BUS_COCO_COCO_GMC_H

**Comment by npwoods** [comment]
Took another look; am I reading it correctly whereby you only added the SLOT_INTERFACE to the CoCo 1/2 declaration, and the Multi-Pak declaration?  If I'm reading things correctly, you wouldn't be able to mount this in the CoCo 3 driver without using a Multi-Pak

**Comment by tlindner** [comment]
You're right. It's isn't working under the CoCo 3. I thought the CoCo 3 inherited the slot devices from the CoCo, and never tested. What is the correct way to add this to the coco3?

**Comment by tlindner** [comment]
Oops. It's working fine on the CoCo 3 driver. I had the Becker port turned on and heard no sound. But once I turned it off everything was fine. Here is the funky command I must use with the multi pack:

./mame64 coco3 -ext multi -ext:multi:slot4 games_master -cart5 cocofest.rom

**Comment by npwoods** [comment]
Can you get it to work without the Multi-Pak?  If I'm reading the source correctly, you probably cannot.

On that note, it is probably bad that these slot declarations have to be copied and pasted around like this.

**Comment by tlindner** [comment]
Yup. Work with and without multipak, CoCo 1/2 and 3.

**Comment by cuavas** [comment]
This can't possible pass validation tests.  Please fix your device constructor.

**Pull request merged**